backend/internal/core/services/deployment_service.go

```diff
@@ -229,7 +235,20 @@ func (svc *DeploymentService) handleDeploymentAction(userID int, workflowName st
 		return "", "", err
 	}
 
+	if len(nodeIDs) > 0 {
+
+		if err = svc.locker.AcquireWorkflowLock(svc.appCtx, nodeIDs, wf.UUID); err != nil {
```

Comment:
not sure what's the point of workflow lock. it's a single worker deploying, right? and the nodes are locked in the handler?